### PR TITLE
Showcase

### PR DESCRIPTION
**Pull request by NYBACHOK**

### CODE REVIEW

File: abci/src/server.rs [hunk 1]
```diff
@@ -73,13 +76,21 @@ pub struct Server<App> {
...
-            let (stream, addr) = self.listener.accept().map_err(Error::io)?;
-            let addr = addr.to_string();
-            info!("Incoming connection from: {}", addr);
-            self.spawn_client_handler(stream, addr);
-        }
+    pub fn listen(self) -> Result<(), Error> { 
+
+        let _ = thread::spawn( move || {
+            loop {
+                let (stream, addr) = self.listener.accept().map_err(Error::io).expect( "failed to accept connection");
```
Comment (joneskm) [inline abci/src/server.rs]: Why do we `expect` now instead of returning the error?

File: abci/src/server.rs [hunk 2]
```diff
@@ -73,13 +76,21 @@ pub struct Server<App> {
...
+                let addr = addr.to_string();
+                info!("Incoming connection from: {}", addr);
+                self.spawn_client_handler(stream, addr);
+            }
+           
+        });
+        
+        while !CancellationSource::is_cancelled() { }
```
Comment (joneskm) [inline abci/src/server.rs]: Isn't this going to consume the CPU?

Comment (NYBACHOK) [reply abci/src/server.rs]: Yes. I need to add `thread::sleep`

File: abci/src/server.rs (hunk 1 above)
Comment (NYBACHOK) [reply abci/src/server.rs]: 'Cause of [this line ](https://github.com/rumos-io/tendermint-rs/blob/f7b0a5a4b38dfbb51b51d051907154bc916800a2/abci/src/server.rs#L83C16-L83C119) of code. It blocks the main thread and I chose to move it into a separate thread. 
What do you think:
1. Call expect on error
2. Send error with mpsc
3. Ignore error and log it
4. Don't panic but request cancellation

Comment (NYBACHOK) [reply abci/src/server.rs]: Personally I think. Approach with logging error and stopping application better

Comment (joneskm) [reply abci/src/server.rs]: Agreed

**Pull request closed without merging**